Lib/test/test_pydoc/test_pydoc.py

```diff
@@ -2161,10 +2161,47 @@ def test_url_requests(self):
 
 
 class TestHelper(unittest.TestCase):
+    def mock_interactive_session(self, inputs):
+        """
+        Given a list of inputs, run an interactive help session.  Returns a string
+        of what would be shown on screen.
+        """
+        input_iter = iter(inputs)
+
+        def mock_getline(prompt):
+            output.write(prompt)
+            next_input = next(input_iter)
+            output.write(next_input + os.linesep)
+            return next_input
+
+        with captured_stdout() as output:
+            helper = pydoc.Helper(output=output)
+            with unittest.mock.patch.object(helper, "getline", mock_getline):
+                helper.interact()
+
+        # handle different line endings across platforms consistently
+        return output.getvalue().splitlines(keepends=False) + ['']
```

Comment:
Can we avoid some of the empty-string weirdness in the test cases?  Something like the following?

```suggestion
        return output.getvalue().strip().splitlines(keepends=False)
```

(or `.strip('\r\n')` if we want to keep trailing spaces for some reason).